File: memory-domain/src/main/java/com/memory/dto/map/response/MapResponse.java
package com.memory.dto.map.response;

import com.memory.domain.map.Map;
import com.memory.domain.map.MapType;
import com.memory.dto.member.response.MemberResponse;

public record MapResponse(
        Long id,
        String name,
        String description,
        String address,
        String latitude,
        String longitude,
        MapType mapType,
        MemberResponse member
) {

    public static MapResponse from(Map map) {
        return new MapResponse(
                map.getId(),
                map.getName(),
                map.getDescription(),
                map.getAddress(),
                map.getLatitude(),
                map.getLongitude(),
                map.getMapType(),
                MemberResponse.from(map.getMember())
        );
    }
}

File: memory-api/src/main/kotlin/com/memory/dto/map/response/MapResponse.kt
package com.memory.dto.map.response

import com.memory.domain.map.Map
import com.memory.domain.map.MapType
import com.memory.dto.member.response.MemberResponse

data class MapResponse(
    val id: Long?,
    val name: String?,
    val description: String?,
    val address: String?,
    val latitude: String?,
    val longitude: String?,
    val mapType: MapType?,
    val member: MemberResponse?
) {
    companion object {
        @JvmStatic
        fun from(map: Map): MapResponse {
            return MapResponse(
                map.id,
                map.name,
                map.description,
                map.address,
                map.latitude,
                map.longitude,
                map.mapType,
                MemberResponse.from(map.member)
            )
        }
    }
}